File: typescript/7kyu/middle_character/middle_character.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Challenge = void 0;
class Challenge {
    static getMiddle(s) {
        return s.substring((s.length - 1) / 2, s.length / 2 + 1);
    }
}
exports.Challenge = Challenge;
const chai_1 = require("chai");
function test(string, expected) {
    chai_1.assert.strictEqual(Challenge.getMiddle(string), expected);
}
describe("solution", function () {
    it("should handle basic tests", function () {
        test("test", "es");
        test("testing", "t");
    });
});

File: dart/7kyu/middle_character/bin/middle_character.dart
// See https://pub.dartlang.org/packages/test
import "package:test/test.dart";

void main() {
  test("test", () {
    expect(getMiddle("test"), equals("es"));
  });
  test("testing", () {
    expect(getMiddle("testing"), equals("t"));
  });
  test("middle", () {
    expect(getMiddle("middle"), equals("dd"));
  });
  test("A", () {
    expect(getMiddle("A"), equals("A"));
  });
}

String getMiddle(String s) {
  return s.substring((s.length - 1) ~/ 2, s.length ~/ 2 + 1);
}
